plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions/AddToCartWithOptions.php

```diff
@@ -166,15 +166,34 @@ protected function render( $attributes, $content, $block ) {
 			 */
 			$default_quantity = apply_filters( 'woocommerce_add_to_cart_quantity', 1, $product->get_id() );
 
+			wp_interactivity_state(
+				'woocommerce/add-to-cart-with-options',
+				array(
+					'isFormValid'          => ! $product->is_type( 'variable' ),
+					'productOrVariationId' => $product->get_id(),
+				)
+			);
+
 			$context = array(
 				'productId'   => $product->get_id(),
 				'productType' => $product->get_type(),
 				'quantity'    => $default_quantity,
 			);
 
 			if ( $product->is_type( 'variable' ) ) {
+				$context['variationId']         = null;
 				$context['selectedAttributes']  = array();
-				$context['availableVariations'] = $product->get_available_variations();
+				$available_variations           = $product->get_available_variations();
+				$available_variations_data      = array_map(
+					function ( $variation ) {
+						return array(
+							'variation_id' => $variation['variation_id'],
+							'attributes'   => $variation['attributes'],
+						);
+					},
+					$available_variations
+				);
+				$context['availableVariations'] = $available_variations_data;
```

Comment:
There is no need to print the whole data for available variations as in the frontend we only read `variation_id` and `attributes`. While this is not directly required for this PR, I decided to include it here after the discussion in #57727.